src/core/config/transformer/improve_type_names.rs

```diff
@@ -61,48 +62,66 @@ impl<'a> CandidateConvergence<'a> {
-                if self.config.is_scalar(&field_info.type_of) {
-                    // If field type is scalar then ignore type name inference.
-                    continue;
-                }
+        // process the user suggest field names over the auto inferred names.
+        let ty_names = vec![
+            self.config.schema.query.as_ref(),
+            self.config.schema.mutation.as_ref(),
+            self.config.schema.subscription.as_ref(),
+        ]
+        .into_iter()
+        .flatten()
+        .map(|o| o.to_owned())
+        .chain(self.config.types.keys().cloned())
+        .collect::<IndexSet<_>>();
+
+        for ref type_name in ty_names {
+            if let Some(type_info) = self.config.types.get(type_name) {
+                for (field_name, field_info) in type_info.fields.iter() {
+                    if self.config.is_scalar(&field_info.type_of) {
+                        // If field type is scalar then ignore type name inference.
+                        continue;
+                    }
 
-                let inner_map = self
-                    .candidates
-                    .entry(field_info.type_of.to_owned())
-                    .or_default();
-
-                let singularized_candidate = field_name.to_singular();
-
-                if let Some(key_val) = inner_map.get_mut(&singularized_candidate) {
-                    key_val.frequency += 1
-                } else {
-                    // in order to infer the types correctly, always prioritize the non-operation
-                    // types but final selection will still depend upon the
-                    // frequency.
-                    let priority = match self.config.is_root_operation_type(type_name) {
-                        true => 0,
-                        false => 1,
-                    };
-
-                    inner_map.insert(
-                        singularized_candidate,
-                        CandidateStats { frequency: 1, priority },
-                    );
+                    let inner_map = self
+                        .candidates
+                        .entry(field_info.type_of.to_owned())
+                        .or_default();
+
+                    let singularized_candidate = field_name.to_singular();
+
+                    if let Some(key_val) = inner_map.get_mut(&singularized_candidate) {
+                        key_val.frequency += 1
+                    } else {
+                        let priority = match self.config.is_root_operation_type(type_name) {
+                            true => {
+                                if self.suggested_names.contains(field_name) {
+                                    // priority of user suggested name is higher than anything.
+                                    2
+                                } else {
+                                    0
+                                }
+                            }
```

Comment:
Increase the priority if it's a suggested name, instead of setting it to 2 or 0.